R/query-string.R

```diff
@@ -8,29 +8,36 @@ queryStringFilter <- function(req){
 #' @importFrom utils URLdecode
 #' @noRd
 parseQS <- function(qs){
-  if (is.null(qs) || length(qs) == 0 || qs == ""){
+  # Is there data in the request?
+  if (is.null(qs) || length(qs) == 0 || qs == "") {
     return(list())
   }
-  if (stri_startswith_fixed(qs, "?")){
-    qs <- substr(qs, 2, nchar(qs))
-  }
-
-  qs <- gsub("+", " ", qs, fixed=TRUE)
+ # Is it JSON data?
+  if (stri_startswith_fixed(qs, "{")) {
+    # Handle JSON with jsonlite
+    ret <- jsonlite::fromJSON(qs)
+    } else {
```

Comment:
Nit: Looks like the indentation is off here. Mind shifting this back one indent?